code/datums/martial/worldshaker.dm

```diff
@@ -0,0 +1,546 @@
+	no_guns = TRUE
+	help_verb = /mob/living/carbon/human/proc/worldshaker_help
+	block_chance = 100 //validhunters cry
+	var/list/thrown = list()
+	COOLDOWN_DECLARE(next_leap)
+	COOLDOWN_DECLARE(next_balloon)
+	var/datum/action/cooldown/worldstomp/linked_stomp
+	var/leaping = FALSE
+	var/plates = 0
+	var/plate_timer = null
+	var/heavy = FALSE //
+
+/datum/martial_art/worldshaker/can_use(mob/living/carbon/human/H)
+	if(H.stat == DEAD || H.IsUnconscious() || H.IsFrozen() || HAS_TRAIT(H, TRAIT_PACIFISM))
+		return FALSE
+	return ispreternis(H)
+
+/datum/martial_art/worldshaker/disarm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	return TRUE //you're doing enough pushing as is
+
+/datum/martial_art/worldshaker/harm_act(mob/living/carbon/human/A, mob/living/D)
+	return TRUE //no punch, just pummel
+
+/datum/martial_art/worldshaker/proc/InterceptClickOn(mob/living/carbon/human/H, params, atom/target)
+	var/list/modifiers = params2list(params)
+	if(!(can_use(H)) || (modifiers["shift"] || modifiers["alt"] || modifiers["ctrl"]))
+		return
+	if(H.a_intent == INTENT_DISARM)
+		leap(H, target)
+	if(H.a_intent == INTENT_HELP && (H==target))
+		rip_plate(H)
+	if(thrown.len > 0 && H.a_intent == INTENT_GRAB)
+		if(get_turf(target) != get_turf(H))
+			lob(H, target)
+
+	if(!H.Adjacent(target) || H==target)
+		return
+	if(H.a_intent == INTENT_HARM && isliving(target))
+		pummel(H,target)
+	if(H.a_intent == INTENT_GRAB)
+		grapple(H,target)
+
+	
+/*-------------------------------------------------------------
+	start of helpers section
+---------------------------------------------------------*/
+/datum/martial_art/worldshaker/proc/stagger(mob/living/victim)
+	if(HAS_TRAIT(victim, TRAIT_STUNIMMUNE))
+		return
+	victim.set_resting(TRUE)//basically a trip
+	victim.add_movespeed_modifier(id, update=TRUE, priority=101, multiplicative_slowdown = 0.5)
+	addtimer(CALLBACK(src, PROC_REF(stagger_end), victim), STAGGER_DURATION, TIMER_UNIQUE | TIMER_OVERRIDE)
+
+/datum/martial_art/worldshaker/proc/stagger_end(mob/living/victim)
+	victim.remove_movespeed_modifier(id)
+
+/datum/martial_art/worldshaker/proc/push_away(mob/living/user, atom/movable/victim, distance = 1)
+	var/throwdirection = get_dir(user, victim)
+	var/atom/throw_target = get_edge_target_turf(victim, throwdirection)
+	var/throwspeed = 3
+	if(heavy)
+		throwspeed *= 2
+		distance *= 2
+	victim.throw_at(throw_target, distance, throwspeed, user)
+/*---------------------------------------------------------------
+	end of helpers section
+----------------------------------------------------------------*/
+/*---------------------------------------------------------------
+	start of plates section 
+---------------------------------------------------------------*/
+/datum/martial_art/worldshaker/proc/grow_plate(mob/living/carbon/human/user)
+	if(plates >= PLATE_CAP || user.stat == DEAD)//no quaking the entire station
+		return
+	user.balloon_alert(user, span_notice("Your plates grow thicker!"))
```

Comment:
balloon alerts are not used for this purpose, they take short inputs for a quick feedback (and also can't take spans) and what i said in another PR it is good if you don't capitalize unimportant words + use minimum words as possible

Reply:
Ah, gotcha
Yeah, i was completely oblivious to balloon alerts before your changes on that other PR so I definitely went overboard with them.

Also didnt realize it cant take spans